Lib/test/test_capi.py

```diff
@@ -857,8 +857,13 @@ class PyMemDebugTests(unittest.TestCase):
 
     def check(self, code):
         with support.SuppressCrashReport():
-            out = assert_python_failure('-c', code,
-                                        PYTHONMALLOC=self.PYTHONMALLOC)
+            out = assert_python_failure(
+                '-c', code,
+                PYTHONMALLOC=self.PYTHONMALLOC,
+                # FreeBSD: instruct jemalloc to not fill freed() memory
```

Comment:
this is a rather fragile test as written. depending on the undefined behavior of what the contents of free()d memory will be is our problem.  jemalloc is doing something perfectly reasonable, it's our test that is not. expect this to come up again at a random future point even if this workaround does something useful today.

Reply:
Other memory allocators may use different constants. I have an open PR for mimalloc https://github.com/microsoft/mimalloc/pull/517

Reply:
Maybe _PyMem_IsPtrFreed() constants could be updated at runtime if Python detects when jemalloc is running?

I already had to change _PyMem_IsPtrFreed() constants when I implemented this function, to use Microsoft memory allocator constants.

_PyMem_IsPtrFreed() is only called by _PyObject_IsFreed() which is only used when Python is built in debug mode (and a few _testcapi test functions in release mode as well). So performance is not critical here.

Reply:
You would have to detect that you run under jemalloc **and** jemalloc is built with debug flags **or** jemalloc is configured to taint freed memory. FreeBSD 14.0 is under development and its libc uses a debug build. Release builds of FreeBSD are not affected.

Reply:
A lot of memory allocators do this when not building release/optimized builds. It isn't just jemalloc. I don't see an obvious clean solution.